realm/realm-library/src/main/java/io/realm/RealmResults.java

```diff
@@ -64,45 +65,51 @@
     private TableOrView table = null;
 
     private static final String TYPE_MISMATCH = "Field '%s': type mismatch - %s expected.";
-    private long currentTableViewVersion = -1;
+    private long currentTableViewVersion = 0;
 
     private final TableQuery query;
     private final List<RealmChangeListener> listeners = new CopyOnWriteArrayList<RealmChangeListener>();
     private Future<Long> pendingQuery;
     private boolean isCompleted = false;
 
-    static <E extends RealmObject> RealmResults<E> createFromClass(BaseRealm realm, Class<E> clazz) {
-        return new RealmResults<E>(realm, clazz);
-    }
-
     static <E extends RealmObject> RealmResults<E> createFromTableOrView(BaseRealm realm, TableOrView table, Class<E> clazz) {
         return new RealmResults<E>(realm, table, clazz);
     }
 
-    static RealmResults<DynamicRealmObject> createFromDynamicClass(BaseRealm realm, String className) {
-        return new RealmResults<DynamicRealmObject>(realm, className);
+    static RealmResults<DynamicRealmObject> createFromDynamicClass(BaseRealm realm, TableQuery query, String className) {
+        return new RealmResults<DynamicRealmObject>(realm, query, className);
     }
 
     static RealmResults<DynamicRealmObject> createFromDynamicTableOrView(BaseRealm realm, TableOrView table, String className) {
         return new RealmResults<DynamicRealmObject>(realm, table, className);
     }
 
-    private RealmResults(BaseRealm realm, Class<E> classSpec) {
+    RealmResults(BaseRealm realm, TableQuery query, Class<E> clazz) {
```

Comment:
Use factory constructor instead. They where introduced because the no-name constructor was far to magical given the number of parameters provided. A named factory constructor make it it a lot easier to read the code elsewhere